src/link.js

```diff
@@ -0,0 +1,73 @@
+// Linter can see that API is used in types.
+// eslint-disable-next-line
+import * as API from "./link/interface.js"
+import { CID, format } from './cid.js'
+// This way TS will also expose all the types from module
+export * from './link/interface.js'
+
+const DAG_PB_CODE = 0x70
+// eslint-disable-next-line
+const SHA_256_CODE = 0x12
+
+/**
+ * Simplified version of `create` for CIDv0.
+ * @param {API.MultihashDigest<typeof SHA_256_CODE>} digest - Multihash.
+ * @returns {API.LegacyLink}
+ */
+export const createLegacy = digest => CID.create(0, DAG_PB_CODE, digest)
+
+/**
+ * Simplified version of `create` for CIDv1.
+ * @template {unknown} Data
+ * @template {number} Code
+ * @template {number} Alg
+ * @param {Code} code - Content encoding format code.
+ * @param {API.MultihashDigest<Alg>} digest - Miltihash of the content.
+ * @returns {API.Link<Data, Code, Alg>}
+ */
+export const create = (code, digest) => CID.create(1, code, digest)
+
+/**
+ * Type predicate returns true if value is the link.
+ *
+ * @template {API.Link<unknown, number, number, 0|1>} L
+ * @param {unknown|L} value
+ * @returns {value is L & CID}
+ */
+export const isLink = value =>
```

Comment:
I thought you didn't like `isX()` but preferred an `asX()` pattern?

Reply:
I do like [parse don’t validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/) approach. Yet here it seems different because it’s interface, rather than a concrete class, but I suppose `asLink` would be more consistent.

For what it’s worth, back when I made `isCID` → `asCID` case, typescript did not had [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) or I didn’t know of it’s existence and with `boolean` return type no refinement took place on `x` in`if (CID.isCID(x)) x`. With`if (isLink(x)) x` refinement occurs (thanks to return type been `x is Link`

it is also that `isCID` actually could return true, even if `x` was of different type (like CID from previous version) possibly  with less methods, but return would have been `true`. With `Link` however version doesn’t matter so `true` seems more accurate.

I’m happy to change to `asLink`

Reply:
> `if (isLink(x)) x` refinement occurs

makes sense

From a JS consumer perspective though, we've encouraged the `asCID` pattern already so maybe we should continue with that. Under your influence I find myself being dragged in the _parse don't validate_ direction, not even using TS so this feels like a backward step when not using TS.